src/Tizen.NUI.Components/Controls/AlertDialog.cs

```diff
@@ -175,425 +116,369 @@ public View TitleContent
-                }
+                message = value;
 
-                popupContent = value;
-                if (popupContent == null)
+                if (Content is TextLabel textLabel)
                 {
-                    return;
+                    textLabel.Text = message;
                 }
-
-                ResetContent();
             }
         }
 
         /// <summary>
-        /// Action content of AlertDialog. ActionContent is added to Children automatically.
+        /// Content of AlertDialog. Content is added to Children automatically.
         /// </summary>
         [EditorBrowsable(EditorBrowsableState.Never)]
-        public View ActionContent
+        public View Content
         {
             get
             {
-                return popupAction;
+                return content;
             }
             set
             {
-                if (popupAction == value)
+                if (content == value)
                 {
                     return;
                 }
 
-                if (popupAction != null)
+                if (content != null)
                 {
-                    Remove(popupAction);
+                    Remove(content);
                 }
 
-                popupAction = value;
-                if (popupAction == null)
+                content = value;
+                if (content == null)
                 {
                     return;
                 }
 
-                ResetContent();
-            }
-        }
-
-        private void ResetContent()
-        {
-            if (Content == null)
-            {
-                return;
-            }
-
-            //To keep the order of TitleContent, BodyContent and ActionContent,
-            //the existing contents are removed and added again.
-            if (popupTitle != null)
-            {
-                Content.Remove(popupTitle);
-            }
-
-            if (popupContent != null)
-            {
-                Content.Remove(popupContent);
-            }
-
-            if (popupAction != null)
-            {
-                Content.Remove(popupAction);
-            }
-
-            if (popupTitle != null)
-            {
-                Content.Add(popupTitle);
-            }
-
-            if (popupContent != null)
-            {
-                Content.Add(popupContent);
-            }
+                if (content is TextLabel textLabel)
+                {
+                    textLabel.Text = message;
+                }
 
-            if (popupAction != null)
-            {
-                Content.Add(popupAction);
+                ResetContent();
             }
         }
 
         /// <summary>
-        /// Title text of AlertDialog.
+        /// Action views of AlertDialog.
+        /// Action views are added to ActionContent of AlertDialog.
         /// </summary>
         [EditorBrowsable(EditorBrowsableState.Never)]
-        public string Title
+        public IEnumerable<View> Actions
```

Comment:
I think it is very good to use IEnumerable as a return type. is it of the purpose not to use `array[]` as a return type?
I am not sure, but if IEnumerable is used, IEnumerator should be defined. isn't it? I don't know very well so I am just asking to figure out together. ^^;
I saw this : https://docs.microsoft.com/en-us/dotnet/api/system.collections.ienumerable?view=net-5.0

Reply:
I was guided by https://github.com/Samsung/TizenFX/pull/2861#discussion_r609570342 and https://github.com/Samsung/TizenFX/pull/2861#discussion_r609643301

The above comments refer https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/guidelines-for-collections and it says ArrayList or List<T> are not recommended because it does not notice when user modifies the collections.

So IEnumerable can be used instead of ArrayList or List<T>.

Reply:
OK, I see. Thank you for explanation.